hook.sh

```diff
@@ -46,17 +46,40 @@ echo -e "\e[32mSetting challenge to ${CERTBOT_VALIDATION} ...\e[0m"
 domain=.$CERTBOT_DOMAIN
 infix=${domain%.$DEDYN_NAME}
 
+# Remove leading wildcard from infix, if present
+# *.foobar.dedyn.io gives "" while *.a.foobar.dedyn.io gives ".a"
+infix=${infix#.\*}
+
 args=( \
-    '-Ss' \
+    '-sSLf' \
     '-H' "Authorization: Token $DEDYN_TOKEN" \
     '-H' 'Accept: application/json' \
     '-H' 'Content-Type: application/json' \
-    '-d' '[{"subname":"_acme-challenge'"$infix"'", "type":"TXT", "records":["\"'"$CERTBOT_VALIDATION"'\""], "ttl":60}]' \
-    '-o' '/dev/null' \
 )
 
+# Letsencrypt uses a TTL of 60 seconds on their resolvers. If we're getting
```

Comment:
Does LE really use a 60 seconds TTL on their servers? My hypothesis is that they always query the authoritative servers, so that the TTL does not play any role, and the only relevant timing is the replication time for the DNS slaves (but I'm not sure about that). Do you have a reference for your statement?

Reply:
The mechanics aren't super well-documented, but I was going off of this comment from one of the LE engineers: https://community.letsencrypt.org/t/dns-servers-used-by-letsencrypt-for-challenges/32127/10

In my test case, publishing 2 challenges for an ECDSA certificate (and verifying they were available before returning to certbot) succeeded, but immediately afterwards publishing 2 challenges (for the same domain) for an RSA certificate would fail, even if I verified i could `dig` those records `@ns1.desec.io`. I was assuming the old records were still stuck in the Unbound cache cpu mentioned.

Any idea what the typical propagation time is between the slaves? I can play with this value somewhat and save people the wait.

Reply:
The propagation time between our API responding with a success code and the slaves actually serving that content is usually a few seconds, at least in Europe. In a few percent of cases, however, it can take up to around one minute.

We spent a lot of effort on getting it really fast and reliable, but it's actually a very tricky issue. We hope to improve it further in the future ...